src/lib.rs

```diff
@@ -1,37 +1,304 @@
-use lazy_static::lazy_static;
-use std::process;
+//! This crate aims to provide an easy and customizable way to help you build
+//! Wasm projects by extending them with custom subcommands, based on the
+//! [`xtask` concept](https://github.com/matklad/cargo-xtask/), instead of using
+//!  external tooling like [`wasm-pack`](https://github.com/rustwasm/wasm-pack).
+//!
+//! # Minimum Supported Rust Version
+//!
+//! This crate requires **Rust 1.58.1** at a minimum.
+//!
+//! # Setup
+//!
+//! The best way to add xtask-wasm to your project is to create a workspace
+//! with two packages: your project's package and the xtask package.
+//!
+//! ## Create or update a project using xtask
+//!
+//! * New project - You can create a new project using xtask, you can use the
+//!     following:
+//!     ```console
+//!     mkdir project_name
+//!     cd project_name
+//!     cargo new project
+//!     cargo new xtask
+//!     touch Cargo.toml
+//!     ```
+//!     Open the workspace's Cargo.toml and add the following:
+//!     ```toml
+//!     [workspace]
+//!     members = [
+//!         "project",
+//!         "xtask",
+//!     ]
+//!     ```
+//!
+//! * Project with a single package - If you project contains only one package,
+//!     move all the content of the project expect the `.git` directory into a
+//!     new directory named after the package name at the root of the project.
```

Comment:
No you don't need to do that actually. You can have a root package and a workspace.